Mathlib/RingTheory/LittleWedderburn.lean

```diff
@@ -0,0 +1,162 @@
+  -- The key thing is then to note that for all conjugacy classes `x`, `|x|` is given by
+  -- `|Dˣ| / |Zxˣ|`, where `Zx` is the centralizer of `x`; but `Zx` is an algebra over `Z`, and
+  -- therefore `|Zxˣ| = q ^ d - 1`, where `d` is the dimension of `D` as a vector space over `Z`.
+  -- We therefore get that `|x| = (q ^ n - 1) / (q ^ d - 1)`, and as `d` is a strict divisor of `n`,
+  -- we do have that `Φₙ(q) | (q ^ n - 1) / (q ^ d - 1)`; extending this over the whole sum
+  -- gives us the desired contradiction..
+  rw [Set.toFinset_card, ConjClasses.card_carrier, ← card_congr
+        (show Zxˣ ≃* _ from unitsCentralizerEquiv _ x).toEquiv, card_units, card_D]
+  have hZx : Zx ≠ ⊤ := by
+    by_contra! hZx
+    refine (ConjClasses.mk_bijOn (Dˣ)).mapsTo (Set.subset_center_units ?_) hx
+    refine Subring.centralizer_eq_top_iff_subset.mp hZx <| Set.mem_singleton _
+  letI : Field Zx := hD.field hZx.lt_top
+  letI : Algebra Z Zx := (Subring.inclusion <| Subring.center_le_centralizer {(x : D)}).toAlgebra
+  let d := finrank Z Zx
+  have card_Zx : card Zx = q ^ d := card_eq_pow_finrank
+  have h1qd : 1 ≤ q ^ d := by rw [← card_Zx]; exact card_pos
+  haveI : IsScalarTower Z Zx D := ⟨fun x y z ↦ mul_assoc _ _ _⟩
+  rw [card_units, card_Zx, Int.coe_nat_div, Nat.cast_sub h1qd, Nat.cast_sub h1qn, Nat.cast_one,
+      Nat.cast_pow, Nat.cast_pow]
+  apply Int.dvd_div_of_mul_dvd
+  have aux : ∀ {k : ℕ}, ((X : ℤ[X]) ^ k - 1).eval ↑q = (q : ℤ) ^ k - 1 := by
+    simp only [eval_X, eval_one, eval_pow, eval_sub, eq_self_iff_true, forall_const]
+  rw [← aux, ← aux, ← eval_mul]
+  refine (evalRingHom ↑q).map_dvd (X_pow_sub_one_mul_cyclotomic_dvd_X_pow_sub_one_of_dvd ℤ ?_)
+  refine Nat.mem_properDivisors.mpr ⟨⟨_, (finrank_mul_finrank Z Zx D).symm⟩, ?_⟩
+  rw [← (Nat.pow_right_strictMono hq).lt_iff_lt, ← card_D, ← card_Zx]
+  obtain ⟨b, -, hb⟩ := SetLike.exists_of_lt hZx.lt_top
+  refine card_lt_of_injective_of_not_mem _ Subtype.val_injective (?_ : b ∉ _)
+  rintro ⟨b, rfl⟩
+  exact hb b.2
+
+end InductionHyp
+
+private theorem center_eq_top [Finite D] : Subring.center D = ⊤ := by
+  classical
+  cases nonempty_fintype D
+  induction' hn : Fintype.card D using Nat.strong_induction_on with n IH generalizing D
+  apply InductionHyp.center_eq_top
+  intro R hR x y hx hy
+  suffices : (⟨y, hy⟩ : R) ∈ Subring.center R
+  · rw [Subring.mem_center_iff] at this
+    simpa using this ⟨x, hx⟩
+  let R_dr : DivisionRing R := Fintype.divisionRingOfIsDomain R
+  rw [IH (Fintype.card R) _ R inferInstance rfl]
+  · trivial
+  rw [← hn, ← Subring.card_top D]
+  exact Set.card_lt_card hR
+
+end LittleWedderburn
+
+open LittleWedderburn
+
+/-- A finite division ring is a field. See `Finite.isDomain_to_isField` and
+`Fintype.divisionRingOfIsDomain` for more general statements, but these create data, and therefore
+may cause diamonds if used improperly. -/
+def Finite.divisionRing_to_field (D : Type*) [DivisionRing D] [Finite D] : Field D :=
```

Comment:
Are you sure this cannot be an instance? 
```lean
example (D : Type*) [hD : DivisionRing D] [hF : Finite D] : hD = @Field.toDivisionRing D
    (@Finite.divisionRing_to_field D hD hF) := rfl
```
works, so I don't see any problem.

Reply:
Indeed it can; we're not adding any data here